addon/utils/geometry.js

```diff
@@ -0,0 +1,41 @@
+class Rect {
+  static get zero() {
+    return new Rect();
+  }
+
+  constructor({ top = 0, left = 0, width = 0, height = 0 } = {}) {
+    this.top = void 0;
+    this.left = void 0;
+    this.width = void 0;
+    this.height = void 0;
+    this.top = top;
+    this.left = left;
+    this.width = width;
+    this.height = height;
+  }
+
+  get center() {
+    return {
+      x: this.left + this.width / 2,
+      y: this.top + this.height / 2,
+    };
+  }
+}
```

Comment:
nit: add a line between class/method definitions?

Reply:
This is a direct copy/paste from Shopify. I've left it as is to make upgrades easier, but happy to do if you think it's worth it